tools/fuchsia/executable_action.gni

```diff
@@ -0,0 +1,44 @@
+# Copyright 2021 The Fuchsia Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+import("//build/fuchsia/sdk.gni")
+
+if (host_os == "win") {
+  host_executable_suffix = ".exe"
+} else {
+  host_executable_suffix = ""
+}
+
+template("executable_action") {
+  action(target_name) {
+    assert(defined(invoker.tool), "The executable tool must be specified.")
+    if (defined(invoker.visibility)) {
+      visibility = invoker.visibility
+    }
+    if (defined(invoker.testonly)) {
+      testonly = invoker.testonly
+    }
+
+    script = "//build/gn_run_binary.py"
+
+    host_executable =
+        rebase_path("${invoker.tool}${host_executable_suffix}", root_build_dir)
+
+    if (defined(invoker.deps)) {
+      deps = invoker.deps
+    } else {
+      deps = []
+    }
+
+    if (defined(invoker.inputs)) {
```

Comment:
do we want to give the same treatment to `invoker.args`?

Reply:
Done, and I moved `executable_action.gni` to `//tools/fuchsia` because it's actually not fuchsia-specific I think.